active/Win32/Window_and_Application_API_Spec.md

```diff
@@ -0,0 +1,618 @@
+#### Remarks:
+- **Win32**: The image is set as an Icon of the Window. 
+- **UWP**: Do nothing. The property value is ignored.
+
+### Window.Content property
+Gets or sets the UIElement visual root of the window. 
+
+```CS
+[ContentProperty(“Content”)]
+public UIElement Content { get; set; }
+```
+#### Property Value
+UIElement: The visual root of a window.
+
+#### Remarks
+This property has the ContentProperty attribute so the XAML designer can display the content on the designer.
+
+### Window.Visible property
+Gets a value that reports whether the window is visible. Visibility does not consider the window being behind another window and not visible. It only detects if it is on the screen. This property matches the XamlRoot’s IsHostVisible property.
+
+```CS
+public bool Visible { get; } 
+```
+
+#### Property Value
+bool: **true** if the window is visible; **false** if the window is not visible.
+
+#### Remarks
+- **Win32**: False when the Window is minimized or never displayed. 
+- **UWP**: Keep the same current UWP behavior.
+
+
+### Window.VisibilityChanged Event
+
+Occurs when the value of the Visible property changes. Also, it’s fired when the XamlRoot’s IsHostVisible changes as a consequence of cascade change of the Window’s Visibility.
+
+```CS 
+Window w = new Window();
+window.Content = new TextBlock() { Text = "Hello" };
+window.VisibilityChanged += OnVisibilityChanged;
+```	
+
+```CS
+private void OnVisibilityChanged(object sender, VisibilityChangedEventArgs e)
+{
+   if (e.Visible)
+   {
+      //
+   }
+}
+```
+#### Remarks
+VisibilityChangedEventArgs includes a Boolean Visible property that you should check to determine the visibility state when the event was fired.
+
+### Window.SetTitleBar(UIElement) Method
+Set the custom XAML content that should act as the title bar.
+```CS 
+public void SetTitleBar(UIElement); 
+```
+
+#### Parameters
+UIElement: Custom XAML content that should act as the title bar. To use multiple objects, wrap them in a container element such as one derived from Panel.
+
+#### Remarks
+
+- **Win32**: This method do nothing when running in Win32.
+- **UWP**: Continue using the OS behavior. More info [here](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.window.settitlebar) 
+
+### Window.Title property **[NEW]**
+
+Set or get a string to display on the Title. 
+
+```CS
+public string Title { get; set; }
+```
+#### Sample
+```XML
+```XML
+<Window 
+    x:Class="MainWindow"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    Title="Main Window" 
+>
+    
+</Window>
+```
+
+#### Remarks
+This is not a DependencyProperty.
+
+- **UWP**: This string is not displayed in the Window's Title bar when SetTitleBar method is used or set a not null content. The Title is still used in Alt+Tab scenarios,
+
+
+### Window.Current property
+
+Gets the window of the current thread.
+
+```CS
+public static Window Current { get; }
+```
```

Comment:
IMPORTANT: Absolutely do not do this! There is no "current" window when we support multiple windows on a single thread. Trying to store this and update it "depending on the operation" is a very problematic issue that does not have any good answers. It's been well researched for the last decade as part of the ApplicationView / CoreWindow problems with GetForCurrentView.

Reply:
This is an existing API. The purpose of calling it out in this spec is to add the below documentation which explains that it returns null in Desktop apps for that reason.